extensions/narayana-jta/deployment/src/main/java/io/quarkus/narayana/jta/deployment/NarayanaJtaProcessor.java

```diff
@@ -152,6 +168,43 @@ public void build(NarayanaJtaRecorder recorder,
         recorder.setConfig(transactions);
     }
 
+    @BuildStep
+    @Record(STATIC_INIT)
+    public void allowUnsafeMultipleLastResources(NarayanaJtaRecorder recorder,
+            TransactionManagerBuildTimeConfig transactionManagerBuildTimeConfig,
+            Capabilities capabilities, BuildProducer<LogCleanupFilterBuildItem> logCleanupFilters,
+            BuildProducer<NativeImageFeatureBuildItem> nativeImageFeatures) {
+        switch (transactionManagerBuildTimeConfig.unsafeMultipleLastResources
+                .orElse(UnsafeMultipleLastResourcesMode.DEFAULT)) {
+            case ALLOW -> {
+                recorder.allowUnsafeMultipleLastResources(capabilities.isPresent(Capability.AGROAL), true);
+                // we will handle the warnings ourselves at runtime init when the option is set explicitly
+                logCleanupFilters.produce(
+                        new LogCleanupFilterBuildItem("com.arjuna.ats.arjuna", "ARJUNA012139", "ARJUNA012141", "ARJUNA012142"));
+            }
+            case WARN -> {
+                recorder.allowUnsafeMultipleLastResources(capabilities.isPresent(Capability.AGROAL), false);
+                // we will handle the warnings ourselves at runtime init when the option is set explicitly
+                // but we still want Narayana to produce one warning per offending transaction
+                logCleanupFilters.produce(
+                        new LogCleanupFilterBuildItem("com.arjuna.ats.arjuna", "ARJUNA012139", "ARJUNA012142"));
+            }
```

Comment:
@gsmet I think you can probably implement some sort of `warn-once` like this:

```suggestion
            }
            case WARN_ONCE -> {
                recorder.allowUnsafeMultipleLastResources(capabilities.isPresent(Capability.AGROAL), true);
                // we will handle the warnings ourselves at runtime init when the option is set explicitly
                // but we still want Narayana to produce one warning on the first offending transaction
                logCleanupFilters.produce(
                        new LogCleanupFilterBuildItem("com.arjuna.ats.arjuna", "ARJUNA012139", "ARJUNA012142"));
            }
```

Reply:
I think warning for each transaction will be problematic as you will fill the logs with warnings. I'm not sure this can fly in production (so I'm not sure it should be the default in production).